lib/ElementHandle.js

```diff
@@ -96,7 +96,11 @@ class ElementHandle extends JSHandle {
   }
 
   async focus() {
-    await this.executionContext().evaluate(element => element.focus(), this);
+    await this.executionContext().evaluate(element => {
+      element.focus();
+      if (element.ownerDocument.activeElement !== element)
```

Comment:
`element.focus()` calls the event listeners, which may redirect focus to some other element.

Reply:
Indeed, thank you

Reply:
Maybe you can add an event listener for focus, and see if a focus event was fired after calling .focus.

Reply:
Adding an event listener to element.ownerDocument with useCapture true only fails if there is a previous event listener on the document that calls stopImmediatePropagation and doesn't call preventDefault.

Or we can modify [`DOM.focus`](https://aslushnikov.github.io/vanilla-protocol-viewer/#DOM.focus) to return true if the focus was applied successfully.

Reply:
> Or we can modify DOM.focus to return true if the focus was applied successfully.

I like this